discovery_engine/lib/src/domain/document_manager.dart

```diff
@@ -155,13 +155,25 @@ class DocumentManager {
   /// Fails if [sec] is negative or [id] does not identify an active document.
   Future<void> addActiveDocumentTime(
     DocumentId id,
-    DocumentViewMode mode,
+    DocumentViewMode viewMode,
     int sec,
   ) async {
     if (sec < 0) {
       throw RangeError.range(sec, 0, null);
     }
 
+    if (cfgFeatureStorage) {
```

Comment:
🔧 this block is missing `await _engineStateRepo.save(await _engine.serialize());` after the `_engine.timeSpent()` call and before the return. it's going to be changed with the next story anyways, but we should still make the step-wise changes self-contained.